### PR TITLE
#5640 MongoDB connector first iteration

### PR DESCRIPTION
**Pull request by squakez**
First development iteration of the #5337 epic specifically covering the base support #5640 and #6009

### CODE REVIEW

File: app/connector/mongodb/src/test/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/resources/
```
Comment (oscerd) [inline app/connector/mongodb/src/test/.gitignore]: This can be removed

File: app/pom.xml [hunk 2]
```diff
@@ -77,7 +77,7 @@
     <commons.compress.version>1.18</commons.compress.version>
 
     <!-- Global camel version used everywhere -->
-    <camel.version>2.21.0.fuse-740038</camel.version>
+    <camel.version>2.21.0.fuse-750007</camel.version>
```
Comment (oscerd) [inline app/pom.xml]: Check for the version. There are least two other locations to change

File: app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import org.apache.camel.component.extension.verifier.ResultErrorBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static org.apache.camel.util.CastUtils.cast;
+
+public class MongoConnectorVerifierExtension extends DefaultComponentVerifierExtension {
```
Comment (lburgazzoli) [inline app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java]: unless there is something syndesis specific, this verifier could be contributed upstream

Comment (oscerd) [reply app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java]: +1

Comment (squakez) [reply app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java]: Was not aware of the feature in Camel, could you point me out on some example to follow up?

File: app/pom.xml (hunk 2 above)
Comment (squakez) [reply app/pom.xml]: Extension and Integration BOM as far I can see, thanks for pointing out!

File: app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java (hunk 3 above)
Comment (squakez) [reply app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java]: I created an issue to pull that off Syndesis in a next iteration development: #6280

**Pull request merged**